eth2util/deposit/deposit.go

```diff
@@ -105,6 +105,27 @@ func MarshalDepositData(pubkey eth2p0.BLSPubKey, withdrawalAddr string, network
 	return bytes, nil
 }
 
+// MarshalDepositDatas serializes a list of deposit data into a single file.
+func MarshalDepositDatas(depositDatas [][]byte) ([]byte, error) {
```

Comment:
Don't we need separate files?

Reply:
No. The deposit data file is supposed to be a single file containing an array of deposit datas, one for each validator. That is how even the `staking-deposit-cli` generates. The subsequent PR https://github.com/ObolNetwork/charon/pull/557 changes the Marshal functions.